app/src/main/java/com/framgia/forder/screen/shopupdate/ShopUpdateViewModel.java

```diff
@@ -29,4 +66,119 @@ public void onStop() {
     public void setPresenter(ShopUpdateContract.Presenter presenter) {
         mPresenter = presenter;
     }
+
+    public void onClickChooseOpenTime() {
+        mFlag = FLAG_OPEN_TIME;
+        mDialogManager.showTimePickerDialog();
+    }
+
+    public void onClickChooseAutoRejectTime() {
+        mFlag = FLAG_REJECT_TIME;
+        mDialogManager.showTimePickerDialog();
+    }
+
+    @Override
+    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+        String hourString =
+                hourOfDay < 10 ? mContext.getString(R.string.zero) + hourOfDay : "" + hourOfDay;
+        String minuteString =
+                minute < 10 ? mContext.getString(R.string.zero) + minute : "" + minute;
+        if (mFlag == FLAG_OPEN_TIME) {
+            mOpenTime = hourString + mContext.getString(R.string.two_dot) + minuteString;
+            notifyPropertyChanged(BR.openTime);
+        } else {
+            mTimeAutoReject = hourString + mContext.getString(R.string.two_dot) + minuteString;
+            notifyPropertyChanged(BR.timeAutoReject);
+        }
+    }
+
+    public void checked() {
+        mOpenForever = !mOpenForever;
+    }
+
+    public void onClickUpdateShop() {
+        if (!mPresenter.validateDataInput(mName, mDescription)) {
+            return;
+        }
+        UpdateShopRequest updateShopRequest = new UpdateShopRequest();
+        updateShopRequest.setShopId(mShop.getId());
+        updateShopRequest.setName(mName);
+        updateShopRequest.setDescription(mDescription);
+        updateShopRequest.setTimeAutoClose(mOpenTime);
+        updateShopRequest.setOpenForever(mOpenForever);
+        updateShopRequest.setTimeAutoReject(mTimeAutoReject);
+        mPresenter.onUpdateShop(updateShopRequest);
+    }
+
+    @Bindable
+    public String getName() {
+        return mName;
+    }
+
+    @Bindable
+    public String getOpenTime() {
+        return mOpenTime;
+    }
+
+    @Bindable
+    public boolean getOpenDaily() {
```

Comment:
WARNING: [BooleanGetMethodName] A getX() method which returns a boolean should be named isX()
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#BooleanGetMethodName